### PR TITLE
Fix find banking line process

### PR DESCRIPTION
**Pull request by TruongQuangSB**

### CODE REVIEW

File: java/bundles/org.eclipse.set.basis/src/org/eclipse/set/basis/graph/TopPath.java [hunk 1]
```diff
@@ -32,14 +32,14 @@ public class TopPath {
...
 	 *            length of the edges if the path does not cover the full extent
 	 *            of the edges
-	 * @param offset
-	 *            starting offset from the first TOP_Kante in edges
+	 * @param firstEdgeLength
+	 *            the length of the first edge
 	 */
 	public TopPath(final List<TOP_Kante> edges, final BigDecimal length,
-			final BigDecimal offset) {
+			final BigDecimal firstEdgeLength) {
```
Comment (mariusheine) [inline java/bundles/org.eclipse.set.basis/src/org/eclipse/set/basis/graph/TopPath.java:39]: @TruongQuangSB This constructor is also used in `DirectedPathSearch.java`. Do we need to adjust something there accordingly?

File: java/bundles/org.eclipse.set.feature.plazmodel/src/org/eclipse/set/feature/plazmodel/check/BankValues.xtend [hunk 2]
```diff
@@ -54,6 +54,12 @@ class BankValues extends AbstractPlazContainerCheck implements PlazCheck, EventH
 		].map [
 			val bankLength = ueberhoehungslinieAllg.
 				ueberhoehungslinieLaenge.wert
+			if (IDUeberhoehungA?.value === null ||
+				IDUeberhoehungB?.value === null) {
+				return createError(
+					"Die Überhöhungslinie {GUID} fehlt Ueberhoehung_A oder Ueberhoehung_B.",
```
Comment (mariusheine) [inline java/bundles/org.eclipse.set.feature.plazmodel/src/org/eclipse/set/feature/plazmodel/check/BankValues.xtend:60]:
```suggestion
					"Die Überhöhungslinie {GUID} hat entweder keine Ueberhoehung_A oder keine Ueberhoehung_B.",
```

File: java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/AsDirectedTopGraph.java [hunk 3]
```diff
@@ -303,23 +304,30 @@ private static List<DirectedTOPEdge<Edge>> findRelevantEdgesFromTarget(
...
 				final BigDecimal remainingWeight = currentRemaining
 						.subtract(weight);
-
+				final Node edgeSource = graph.getEdgeSource(edge);
 				if (remainingWeight.compareTo(BigDecimal.ZERO) < 1) {
 					continue;
 				}
 				edgesDistance.add(edge);
 				// Mark the edge's source vertex if needed
-				final Node edgeSource = graph.getEdgeSource(edge);
```
Comment (mariusheine) [inline java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/AsDirectedTopGraph.java:318]: @TruongQuangSB Is there any particular reason why you have moved this line up? Any internal assertions?

File: java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/AsDirectedTopGraph.java [hunk 4]
```diff
@@ -303,23 +304,30 @@ private static List<DirectedTOPEdge<Edge>> findRelevantEdgesFromTarget(
...
 				remainingWeigthFromEnd.computeIfAbsent(edgeSource, v -> {
 					nodesToProcess.add(edgeSource);
 					return remainingWeight;
 				});
+
+				// When exist another Edge, which shorter als founded Edge, then
+				// replace the weight
```
Comment (mariusheine) [inline java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/AsDirectedTopGraph.java:326]:
```suggestion
				// When another edge exists, which is shorter than found edge, replace the weight
```

**Review by mariusheine — CHANGES_REQUESTED** [review]

File: java/bundles/org.eclipse.set.basis/src/org/eclipse/set/basis/graph/TopPath.java (hunk 1 above)
Comment (TruongQuangSB) [reply java/bundles/org.eclipse.set.basis/src/org/eclipse/set/basis/graph/TopPath.java:39]: @mariusheine y, u are right. This should be also modified

**Review by mariusheine — APPROVED** [review]

**Pull request merged**